xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp

```diff
@@ -19,29 +19,46 @@
  */
 
 #include "ProcessInfo.h"
-#include "ServiceBroker.h"
 #include "cores/DataCacheCore.h"
 #include "threads/SingleLock.h"
 
-// Override for platform ports
-#if !defined(PLATFORM_OVERRIDE_VP_PROCESSINFO)
+CCriticalSection createSection;
+std::map<std::string, CreateProcessControl> CProcessInfo::m_processControls;
 
-CProcessInfo* CProcessInfo::CreateInstance()
+void CProcessInfo::RegisterProcessControl(std::string id, CreateProcessControl createFunc)
 {
-  return new CProcessInfo();
-}
+  CSingleLock lock(createSection);
 
-#endif
+  m_processControls.clear();
+  m_processControls[id] = createFunc;
+}
 
+CProcessInfo* CProcessInfo::CreateInstance()
+{
+  CSingleLock lock(createSection);
+  
+  CProcessInfo *ret = nullptr;
+  for (auto &info : m_processControls)
+  {
+    ret = info.second();
+    if (ret)
+      return ret;
+  }
+  if (!ret)
```

Comment:
why doing this check here?
This produces a warning that the function end may be reached without a `CProcessInfo*` return.